File: frontend/src/services/comboService.ts
import axiosClient from "./axiosClient";
import { IComboRequest, IComboResponse } from "@/types/combo";
import { ApiResponse, PageResponse, PaginationParams } from "@/types/base";

const API_URL = "/combos";

const comboService = {
    createCombo: async (req: IComboRequest): Promise<ApiResponse<null>> => {
        const res = await axiosClient.post(API_URL, req);
        return res.data;
    },
    updateCombo: async (
        id: number,
        req: IComboRequest,
    ): Promise<ApiResponse<null>> => {
        const res = await axiosClient.put(`${API_URL}/${id}`, req);
        return res.data;
    },
    getComboById: async (id: number): Promise<ApiResponse<IComboResponse>> => {
        const res = await axiosClient.get(`${API_URL}/${id}`);
        return res.data;
    },
    getAllCombos: async (
        keyword: string = "",
        branchId: number | null | undefined,
        isActive: boolean | null = null,
        params: PaginationParams,
    ): Promise<ApiResponse<PageResponse<IComboResponse>>> => {
        const res = await axiosClient.get(API_URL, {
            params: {
                keyword: keyword,
                branchId: branchId,
                isActive: isActive,
                page: params.page ?? 0,
                size: params.size ?? 10,
                sort: `${params.sortBy ?? "createdAt"},${params.sortDirection ?? "desc"}`,
            },
        });
        return res.data;
    },
    deleteCombo: async (id: number): Promise<ApiResponse<null>> => {
        const res = await axiosClient.delete(`${API_URL}/${id}`);
        return res.data;
    },
};

export default comboService;

File: backend/src/main/java/com/mhbilliards/billiards_management/service/combo/ComboService.java
package com.mhbilliards.billiards_management.service.combo;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.mhbilliards.billiards_management.dto.combo.ComboResponseDTO;
import com.mhbilliards.billiards_management.dto.combo.CreateComboDTO;
import com.mhbilliards.billiards_management.dto.combo.UpdateComboDTO;

public interface ComboService {
    ComboResponseDTO createCombo(CreateComboDTO request);

    ComboResponseDTO getComboById(Long id);

    Page<ComboResponseDTO> searchCombos(String keyword, Long branchId, Boolean isActive, Pageable pageable);

    List<ComboResponseDTO> getCombosByBranch(Long branchId);

    ComboResponseDTO updateCombo(Long id, UpdateComboDTO request);

    void deleteCombo(Long id);
}
